client/src/modules/journal/journal.js

```diff
@@ -358,13 +362,17 @@ function JournalController(
   }
 
   // display the journal printable report of selected transactions
-  vm.openJournalReport = function openJournalReport() {
+  vm.openJournalReport = function openJournalReport(type) {
     var url = '/reports/finance/journal';
-    var params = formatExportParameters('pdf');
+    var params = formatExportParameters(type);
 
     if (!params) { return; }
 
-    Modal.openReports({ url : url, params : params });
+    if (type === 'xlsx') {
+      window.location = url + '?'+ $.param(params);
```

Comment:
Hmm... this looks like a hack.  Why do you have to do this?  Can't you do the same thing with an `ng-href` and never call `openJournalReport()`?

Reply:
@jniles I tried to see how you export the journal as  CSV, the operation is done at the client side as can see right?

Reply:
I have a problem, I should know this value 
 var params = formatExportParameters(type);
before sending the request to the server, do you think it is  also possible  even if I use ng-href?

Reply:
Not in all places - that's why @sfount opened #1769.  [Here is an example of the ng-href version](https://github.com/IMA-WorldHealth/bhima-2.X/blob/master/client/src/modules/invoices/registry/registry.html#L43).

Reply:
ok, let me try it out